### ISSUE TITLE
Crash with equalizer and "Add Point Button"

### ISSUE THREAD

**Issue by TooWaBoo** [Beep6581/RawTherapee#3813]
How to reproduce.

1. Open an image and set profile "Neutral"
2. Acivate any equalizer curve and click on "Add Point Button"
![unbenannt](https://cloud.githubusercontent.com/assets/13800920/24832473/8d2c2fce-1cb1-11e7-9aec-a25ab394289b.PNG)

3. Left click and right click in the preview a couple of times.
4. Click on the "Add Point Button" -> Crash

```
Version: 5.0-r1-gtk3-395-g73e14702
Branch: dev
Commit: 73e14702
Commit date: 2017-04-08
Compiler: gcc 6.3.0
Processor: undefined
System: Windows
Bit depth: 64 bits
Gtkmm: V3.22.0
Build type: release
Build flags: -mwin32 -m64 -mthreads -msse2 -std=c++11 -march=native -Werror=unused-label -fopenmp -Werror=unknown-pragmas -Wall -Wno-unused-result -Wno-deprecated-declarations -mwindows -Wno-aggressive-loop-optimizations -DNDEBUG -O3
Link flags: -m64 -mthreads -static-libgcc -march=native -mwindows -s -O3
OpenMP support: ON
MMAP support: ON
```

**Comment by heckflosse** [comment]
Confirmed. It crashes even without 4.

**Comment by agriggio** [comment]
I see this too, on Linux. I'll investigate

**Comment by heckflosse** [comment]
I got a popup with this:

```
(rawtherapee.exe:5384): glibmm-ERROR **: 
unhandled exception (type std::exception) in signal handler:
what: vector::_M_range_check: __n (which is 18446744073709551615) >= this->size() (which is 8)
```

**Comment by agriggio** [comment]
I've just pushed a candidate fix (directly on `dev`). Please test to confirm that it works, thanks!

**Comment by heckflosse** [comment]
@agriggio Because I'm currently working on cppcheck3 branch I applied your fix to my local cppcheck3 to test it.
No more crash :+1:

**Comment by TooWaBoo** [comment]
Great job. Issue can be closed. 👍

**Comment by agriggio** [comment]
fixed in 8f1e2cee42a7ace6d2ac8e6fa35d4089d45831c1
